symcrypt-sys/build/static_link.rs

```diff
@@ -0,0 +1,422 @@
+
+    // Required Windows bcrypt dependency for BCryptGenRandom
+    const ADDITIONAL_DEPENDENCIES: &[&str] = &[
+        #[cfg(windows)]
+        "bcrypt",
+    ];
+    println!("cargo:rerun-if-changed=upstream");
+    println!("Compiling SymCrypt...");
+
+    // Compile and Build SymCrypt with provided SymCryptOptions
+    compile_symcrypt_static(LIB_NAME, &options)?;
+    println!("cargo:rustc-link-lib=static={LIB_NAME}");
+
+    // Link additional dependencies
+    for dep in ADDITIONAL_DEPENDENCIES {
+        println!("cargo:rustc-link-lib=dylib={dep}");
+    }
+
+    Ok(())
+}
+
+// TODO: update -symcrypt_fips_build comment
+
+/// Holds configuration options for compiling SymCrypt.
+///
+/// - `triple`: The target triple
+/// - `symcrypt_use_asm`: Whether to enable assembly optimizations.
+/// - `symcrypt_fips_build`: Whether to build in FIPS mode PLACEHOLDER
+/// - `preconfiged_cc`: Returns a Pre-configured `cc` object for the target triple.
+#[derive(Debug)]
+struct SymCryptOptions {
+    triple: Triple,
+    symcrypt_use_asm: bool,
+    //symcrypt_fips_build: bool, // TODO: Determine if we should expose FIPS build option?
+}
+
+impl SymCryptOptions {
+    fn new() -> Self {
+        Self {
+            triple: Triple::get_target_triple(),
+            symcrypt_use_asm: false, // FIXME: Turn this to true when we get ASM checked in
+                                     //symcrypt_fips_build: false, // TODO: Determine if we should expose FIPS build option?
+        }
+    }
+    fn use_asm(&self) -> bool {
+        self.symcrypt_use_asm
+    }
+    fn triple(&self) -> Triple {
+        self.triple.clone()
+    }
+
+    // Returns a cc object that has been preconfigured for the target triple
+    fn preconfigure_cc(&self) -> cc::Build {
+        let mut cc = cc::Build::new();
+        cc.target(self.triple.to_triple())
+            .include("inc")
+            .include("symcrypt/inc")
+            .include("symcrypt/lib")
+            .warnings(false); // Ignore noisy warnings from SymCrypt
+
+        if !self.symcrypt_use_asm {
+            cc.define("SYMCRYPT_IGNORE_PLATFORM", None);
+        }
+
+        // Set specific flags for each target
+        match self.triple {
+            Triple::x86_64_pc_windows_msvc => {
+                cc.asm_flag("/DSYMCRYPT_MASM"); // TODO: figure out what these flags do
+            }
+            Triple::aarch64_pc_windows_msvc => {
+                cc.define("_ARM64_", None);
+            }
+            Triple::x86_64_unknown_linux_gnu => {
+                cc.include("symcrypt/modules/linux/common"); // TODO: remove?
+                cc.flag("-mpclmul");
+                cc.flag("-Wno-incompatible-pointer-types");
+                /*
+                cc.flag("-mpclmul")
+                    .flag("-mssse3")
+                    .flag("-mxsave")
+                    .flag("-maes")
+                    .flag("-msha")
+                    .flag("-mrdrnd")
+                    .flag("-mrdseed");
```

Comment:
Why comment these out? The flags should match the platform-specific flags SymCrypt is normally build with: [https://github.com/microsoft/SymCrypt/blob/53be637dab201a4c9d95e1cf58040c85d71cf3c2/cmake-configs/SymCrypt-Platforms.cmake#L75](https://github.com/microsoft/SymCrypt/blob/53be637dab201a4c9d95e1cf58040c85d71cf3c2/cmake-configs/SymCrypt-Platforms.cmake#L75)